vm/vmExpr.go

```diff
@@ -699,18 +700,32 @@ func invokeExpr(expr ast.Expr, env *Env) (reflect.Value, error) {
 		return reflect.MakeChan(reflect.ChanOf(reflect.BothDir, t), size), nil
 
 	case *ast.ChanExpr:
-		rhs, err := invokeExpr(e.Rhs, env)
+		rhs, err := invokeExpr(e.Rhs, env, ctx)
 		if err != nil {
 			return nilValue, newError(e.Rhs, err)
 		}
 
 		if e.Lhs == nil {
 			if rhs.Kind() == reflect.Chan {
-				rv, _ := rhs.Recv()
+				var rv reflect.Value
+				cases := []reflect.SelectCase{{
+					Dir:  reflect.SelectRecv,
+					Chan: reflect.ValueOf(ctx.Done()),
+					Send: reflect.ValueOf(nil),
+				}, {
+					Dir:  reflect.SelectRecv,
+					Chan: rhs,
+					Send: reflect.ValueOf(nil),
+				}}
+				var chosen int
+				chosen, rv, _ = reflect.Select(cases)
```

Comment:
Could this be changed to `chosen, rv, _ := reflect.Select(cases)` ?

Reply:
Now that we don't have "interrupt", yes !